wapitiCore/net/crawler_configuration.py

```diff
@@ -17,7 +16,7 @@ class CrawlerConfiguration:
     proxy: str = None
     auth_credentials: tuple = tuple()
```

Comment:
Wow! There are a lot of improvements!

I think a more complex data structure would be appropriate for this variable. The Tuple does not offer any guarantee on the number of data. Therefore, it is necessary to add explicit checks: 

https://github.com/wapiti-scanner/wapiti/pull/310/files#diff-f5a9d807270fdc42bf1ef8d5ed535674fb8467254f7fe2eadbba7c5bb9c68a59R27-R29

A Namedtuple might be more appropriate here.

This is a minor change and the PR has a bigger scope. Feel free to ignore this suggestion 😄

Reply:
I don't like that `auth_credentials` tuple neither :D 

I will work on it later